test/commands/promote.js

```diff
@@ -249,3 +239,83 @@ Promoting PURPLE to DATABASE_URL on myapp... done
     return expect(cmd.run({app: 'myapp', args: {}, flags: {}}), 'to be rejected with', err)
   })
 })
+
+describe('pg:promote when release phase is present', () => {
+  let api
+
+  const attachment = {
+    name: 'PURPLE',
+    addon: {name: 'postgres-1'},
+    namespace: 'credential:hello'
+  }
+
+  const fetcher = () => {
+    return {
+      attachment: () => attachment
+    }
+  }
+
+  const cmd = proxyquire('../../commands/promote', {
+    '../lib/fetcher': fetcher
+  })
+
+  beforeEach(() => {
+    api = nock('https://api.heroku.com:443')
+    api.get('/apps/myapp/formation').reply(200, [{type: 'release'}])
+    api.get('/apps/myapp/addon-attachments').reply(200, [
+      {name: 'DATABASE', addon: {name: 'postgres-1'}, namespace: 'credential:goodbye'},
+      {name: 'RED', addon: {name: 'postgres-1'}, namespace: 'credential:goodbye'},
+      {name: 'PURPLE', addon: {name: 'postgres-1'}, namespace: 'credential:hello'}
+    ])
+    api.post('/addon-attachments', {
+      name: 'DATABASE',
+      app: {name: 'myapp'},
+      addon: {name: 'postgres-1'},
+      namespace: 'credential:hello',
+      confirm: 'myapp'
+    }).reply(201)
+    cli.mockConsole()
+  })
+
+  afterEach(() => {
+    nock.cleanAll()
+    api.done()
+  })
+
+  it('checks release phase', () => {
+    api.get('/apps/myapp/releases').reply(200, [{id: 1, description: 'Attach DATABASE'}, {id: 2, description: 'Detach DATABASE'}])
+    api.get('/apps/myapp/releases/1').reply(200, {status: 'succeeded'})
+    api.get('/apps/myapp/releases/2').reply(200, {status: 'succeeded'})
+    return cmd.run({app: 'myapp', args: {}, flags: {}})
+        .then(() => expect(cli.stderr, 'to equal', expectedMessage + 'Checking release phase... pg:promote succeeded.\n'))
+  })
+
+  it('checks release phase for detach failure', () => {
+    api.get('/apps/myapp/releases').reply(200, [{id: 1, description: 'Attach DATABASE'}, {id: 2, description: 'Detach DATABASE'}])
+    api.get('/apps/myapp/releases/1').reply(200, {status: 'succeeded'})
+    api.get('/apps/myapp/releases/2').reply(200, {status: 'failed'})
+    return cmd.run({app: 'myapp', args: {}, flags: {}})
+        .then(() => expect(cli.stderr, 'to equal', expectedMessage + 'Checking release phase... pg:promote succeeded. It is safe to ignore the failed undefined release.\n'))
```

Comment:
"...safe to ignore the failed undefined release"? 😉

Reply:
lol ;). fixed